### ISSUE TITLE
Uploaded file not showing in interface in 4.15.0

### ISSUE THREAD

**Issue by snakeninny** [gradio-app/gradio#7132]
### Describe the bug

Uploaded file not showing in interface, as you can see in the screenshot video.
The weird is that it's reproducible in my Ubuntu server, but not in my macbook air

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr
import os

def test(image_file):
    if image_file is None:
        raise gr.Error(f"Forgot to upload?")

    return image_file

demo = gr.Interface(
    title="Test",
    fn=test,
    inputs=[
        gr.File(file_types=["image"], file_count="single"),
    ],
    outputs=[
        gr.File(file_types=["image"], file_count="single"),
    ],
    allow_flagging="never",
    examples=[
        [os.path.join(os.path.dirname(__file__), "男.png")],
    ]
)

gr_version = gr.__version__
print("Gradio version:", gr_version)
demo.launch(debug=True, server_name="0.0.0.0")
```

### Screenshot

https://github.com/gradio-app/gradio/assets/641869/faf9a51f-bf9c-4027-b264-a093dc0d2995

### Logs

```shell
There's no error and no log
```

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Linux
gradio version: 4.15.0
gradio_client version: 0.8.1

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.2.0
fastapi: 0.109.0
ffmpy: 0.3.1
gradio-client==0.8.1 is not installed.
httpx: 0.26.0
huggingface-hub: 0.20.3
importlib-resources: 6.1.1
jinja2: 3.1.2
markupsafe: 2.1.3
matplotlib: 3.8.2
numpy: 1.26.3
orjson: 3.9.12
packaging: 23.2
pandas: 2.2.0
pillow: 10.2.0
pydantic: 2.5.3
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0.1
ruff: 0.1.14
semantic-version: 2.10.0
tomlkit==0.12.0 is not installed.
typer: 0.9.0
typing-extensions: 4.9.0
uvicorn: 0.27.0
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.12.2
httpx: 0.26.0
huggingface-hub: 0.20.3
packaging: 23.2
typing-extensions: 4.9.0
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by csyourui** [comment]
Hi,

Same issue here.

It seems only happend on 4.15.0. After uploading a file on the front end, it flashes briefly but does not display the uploaded file information.

## Reproduction:
```
import gradio as gr


with gr.Blocks() as demo:
    file = gr.File(label="upload files")

if __name__ == "__main__":
    demo.launch()

```
## Screenshot
https://github.com/gradio-app/gradio/assets/23717487/f4498f7f-918f-413e-af52-5c84814e3986

**Comment by abidlabs** [comment]
@csyourui  are you also using Linux?

**Comment by koala1113** [comment]
Hi, I have the same issue on 4.15.0.
When the parameter `file_count` of `gr.File` is set to `"single"`, it causes the uploaded file to display incorrectly.

I think it may be related to PR #6980.
The conditions in [File.svelte](https://github.com/gradio-app/gradio/blob/main/js/file/shared/File.svelte#L23) and [FileUpload.svelte](https://github.com/gradio-app/gradio/blob/main/js/file/shared/FileUpload.svelte#L70) ignore the case where `value` is not an `Array`.

**Comment by ThomasCosyn** [comment]
Hi,
We have the same problem on gradio 4.16.0
Thank for your help !

**Comment by csyourui** [comment]
Chrome on macOS

**Comment by dietrich-r** [comment]
I have the same issue in Gradio 4.16.0 and I'm using Chrome and Edge on Windows.

**Comment by freddyaboulton** [comment]
This was fixed in #7221 and will be out in the next release! Thanks for the patience everyone
